### DISCUSSION TITLE
Quartz schema scripts for different database types vs "Never Write Directly To Quartz's Tables" and Data modelers

### DISCUSSION THREAD

**Discussion by s3YwCf2ZbfJG4SHAfjQMAjtsf** [quartznet/quartznet#2009] [Q&A] [answered]
My company is suggesting that tables we provide customers go through a "Data model", using a tool such as Erwin Data Modeler.

Something is just "Screaming at me", that any attempt to "re-model" your tables is like violating a warranty, even though there is $$$ involved.   Do you have any documentation (for/against) "re-model" Quartz tables into a user's database?

Oracle script file
https://github.com/quartznet/quartznet/blob/35b8532dc005af2fc907b6d2516c2e4bf40f833b/database/tables/tables_oracle.sql

You have this best practice:
https://www.quartz-scheduler.net/documentation/best-practices.html#never-write-directly-to-quartz-s-tables

**Comment by lahma** [comment, marked as answer]
Main reason no to directly access database outside of Quartz is transactions. If you create outside locks this can cause deadlocks etc.

I don't know how Erwin Data Modeler works but if you start modifying schema, maybe just fork Quartz and create a custom build and start maintaining it - it would be quite impossible to decipher bug reports against custom schema and/or provide possible future migration scripts.

**Reply by s3YwCf2ZbfJG4SHAfjQMAjtsf** [reply]
-->  maybe just fork Quartz and create a custom build and start maintaining it - it would be quite impossible to decipher bug reports against custom schema and/or provide possible future migration scripts.

I like this reason the most for "against".   :)

Thanks 👍
